mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.kt

```diff
@@ -436,6 +443,10 @@ class MainActivity : AbstractBaseActivity(), ConnectionFactory.UpdateListener {
             }
             else -> {
                 controller.indicateNoNetwork(getString(R.string.error_network_not_available), false)
+                scheduleRetry {
+                    ConnectionFactory.restartNetworkCheck()
```

Comment:
If this helps, our connection tracking is broken. Do we really want to mask such cases?

Reply:
I just copied the code from https://github.com/openhab/openhab-android/blob/master/mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.kt#L537